### DISCUSSION TITLE
Feature Request: CLI .clone command using Attach feature?

### DISCUSSION THREAD

**Discussion by Alex-Monahan** [duckdb/duckdb#5830] [Ideas]
This is not a pressing issue! I was just thinking about ways to use the awesome new Attach feature, and I think it could power functionality like the SQLite shell's `.clone` capability. Maybe we could attach a new blank database, build the schema, insert the data, then detach that database? The built in code for the CLI seems to depend on some SQLite specific syntax (`INSERT OR IGNORE`, and a rowid even in a view I think?).

https://database.guide/clone-sqlite-database/

```sql
(base) PS duckdb_cli_0_6_2_dev> ./duckdb newdb.db
v0.6.2-dev942 2dcdc24556
Enter ".help" for usage hints.
D .clone anotherdb.db
table... Error 0: Parser Error: syntax error at or near "OR"
LINE 1: INSERT OR IGNORE INTO "table" VALUES(?);
               ^ on [SELECT * FROM "table"]
done
tbl... Error 0: Parser Error: syntax error at or near "OR"
LINE 1: INSERT OR IGNORE INTO "tbl" VALUES(?);
               ^ on [SELECT * FROM "tbl"]
done
Error: (0) Binder Error: Referenced column "rowid" not found in FROM clause!
Candidate bindings: "sqlite_schema.sql", "sqlite_schema.type", "sqlite_schema.name"
LINE 1: ...ite_schema WHERE type='table' ORDER BY rowid DESC
                                                  ^ on [SELECT name, sql FROM sqlite_schema WHERE type='table' ORDER BY rowid DESC]
```

**Comment by Mytherin** [comment]
Good idea - we could certainly implement this using attach. We would just need to update the generated SQL for the `.clone` command.
